src/models/ThemeProvider.js

```diff
@@ -133,18 +53,35 @@ export default class ThemeProvider extends Component<ThemeProviderProps, void> {
       throw new StyledError(8)
     }
 
-    return { ...this.outerTheme, ...(theme: Object) }
+    return { ...outerTheme, ...(theme: Object) }
   }
 
-  publish(theme: Theme | ((outerTheme: Theme) => void)) {
-    this.broadcast.publish(this.getTheme(theme))
+  getContext(theme: (outerTheme: ?Theme) => void, outerTheme?: Theme) {
+    return {
+      theme: this.getTheme(theme, outerTheme),
+    }
```

Comment:
What if we instead of using setting context to `{ theme }` we set it to `theme` without a wrapping object, we won't be able to add any other properties on the context but I don't think we will want that anyways?